File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        last_node = dummy
        while(list1 is not None and list2 is not None):
            if list1.val < list2.val:
                next = list1.next
                last_node.next = list1
                list1.next = None
                last_node = list1
                list1 = next
            else:
                next = list2.next
                last_node.next = list2
                list2.next = None
                last_node = list2
                list2 = next
        if list1 is None:
            last_node.next = list2
        if list2 is None:
            last_node.next = list1
        return dummy.next
            
                
            

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
    dummy := &ListNode{Val:0, Next:nil}
	current := dummy
	for list1 != nil && list2 != nil {
		if list1.Val < list2.Val  {
			next := list1.Next
			current.Next = list1
			list1.Next = nil
			list1 = next
		} else {
			next := list2.Next
			current.Next = list2
			list2.Next = nil
			list2 = next
		}
		current = current.Next
	}

	if list1 != nil {
		current.Next = list1
	}
	if list2 != nil {
		current.Next = list2
	}
	return dummy.Next
}
